lark/exceptions.py

```diff
@@ -109,32 +110,56 @@ def match_examples(self, parse_fn, examples, token_type_match_fallback=False, us
                             candidate = label, False
 
         return candidate[0]
+    
+    def _format_terminals(self, names):
+        if self._all_terminals:
+            if isinstance(self._all_terminals, list):
```

Comment:
A bit of a code smell. If all of Lark needs access to the same information, maybe it's better to just pass it prepared, instead of half-baked.